packages/contracts/contracts/Unirep.sol

```diff
@@ -127,17 +125,21 @@ contract Unirep is IUnirep, zkSNARKHelper, VerifySignature {
 
     /**
      * @dev User signs up by providing an identity commitment. It also inserts a fresh state leaf into the state tree.
-     * if user signs up through an atteser who sets airdrop, Unirep will give the user the airdrop reputation.
+     * An attester may specify an `initBalance` of reputation the user can use in the current epoch
      * @param identityCommitment Commitment of the user's identity which is a semaphore identity.
+     * @param initBalance the starting reputation balance
      */
-    function userSignUp(uint256 identityCommitment) external {
+    function userSignUp(uint256 identityCommitment, uint256 initBalance)
```

Comment:
I think the function is good but I think it should not contain the gst attestation